service/matching/matching_engine.go

```diff
@@ -1031,7 +1033,7 @@ func (e *matchingEngineImpl) DescribeTaskQueuePartition(
 	if request.GetVersions() == nil {
 		return nil, serviceerror.NewInvalidArgument("versions must not be nil, to describe the default queue, pass the default build ID as a member of the BuildIds list")
 	}
-	pm, err := e.getTaskQueuePartitionManager(ctx, tqid.PartitionFromPartitionProto(request.GetTaskQueuePartition(), request.GetNamespaceId()), true)
+	pm, err := e.getTaskQueuePartitionManager(ctx, tqid.PartitionFromPartitionProto(request.GetTaskQueuePartition(), request.GetNamespaceId()), true, loadCauseDescribe)
```

Comment:
maybe better to make a new cause for this?

Reply:
This one is accurate, right? Did you mean the versioning one? Maybe it could be one for "read metadata" (including describe) and "write metadata"? I'm not sure we need one for every individual rpc

Reply:
I think it makes sense to use loadCauseDescribe for both DescribeTaskQueuePartition and DescribeTaskQueue, since one causes the other.

I tweaked some other things a little.